repository/maven.go

```diff
@@ -155,24 +152,42 @@ func (r *Maven) writeSettings() (path string, err error) {
 			path)
 		return
 	}
-	settings := id.Settings
-	settings, err = r.injectProxy(id)
+	defer f.Close()
+	settings, err := r.injectProxy(id)
 	if err != nil {
 		return
 	}
+	settings, err = r.injectCacheDir(settings)
 	_, err = f.Write([]byte(settings))
 	if err != nil {
 		err = liberr.Wrap(
 			err,
 			"path",
 			path)
 	}
-	_ = f.Close()
 	addon.Activity("[FILE] Created %s.", path)
 	return
 }
 
-//
+func (r *Maven) injectCacheDir(settings string) (string, error) {
```

Comment:
If we are going to do chaining more in the future, a way to pass around the mxj object might be worth exploring

Reply:
updated